src/wp-includes/class-wp-scripts.php

```diff
@@ -414,6 +451,93 @@ public function do_item( $handle, $group = false ) {
 		return true;
 	}
 
+	/**
+	 * Check if all of a scripts dependents are deferrable which is required to maintain execution order.
+	 *
+	 * Recursively iterate through all registered scripts and for each script that depends on the passed script, check
+	 * that it is deferrable, meaning it uses the defer strategy and so do any scripts that depend on it.
+	 *
+	 * @since 6.2.0
+	 *
+	 * @param string $handle  The script handle.
+	 * @param array  $visited An array of already visited script handles used to avoid looping recursion.
+	 * @return bool True if all dependents are deferrable, false otherwise.
+	 */
+	private function _all_dependents_are_deferrable( $handle, $visited = array() ) {
+		// If this node was already visited, this script can be deferred and the branch ends.
+		if ( in_array( $handle, $visited, true ) ) {
```

Comment:
note that despite this check, when i write a fully circular dependency test (C depends on B depends on A depends on C), PHP was caught in an infinite loop. I believe that happens already, before these changes, but worth testing further and preventing if possible